### PR TITLE
wireshark: update to 3.0.0.

### PR DESCRIPTION
**Pull request by CameronNemo**

### CODE REVIEW

File: srcpkgs/wireshark/template [hunk 1]
```diff
@@ -1,37 +1,32 @@
...
-hostmakedepends="pkg-config flex perl python-devel mit-krb5-devel automake libtool
- qt5-host-tools qt5-tools glib-devel"
+build_style=cmake
+hostmakedepends="pkg-config flex perl python-devel mit-krb5-devel
+ qt5-host-tools qt5-tools"
 makedepends="libpcap-devel zlib-devel lua52-devel mit-krb5-devel gnutls-devel
- geoip-devel libgcrypt-devel gtk+3-devel qt5-devel libcap-devel sbc-devel c-ares-devel"
+ libmaxminddb-devel libgcrypt-devel libcap-devel sbc-devel c-ares-devel
+ qt5-devel qt5-tools-devel qt5-multimedia-devel qt5-svg-devel gtk+3-devel
+ gtk+3-devel"
```
Comment (jnbr) [inline srcpkgs/wireshark/template]: is `gtk+3-devel` still used? anyway, this one is a duplicate

File: srcpkgs/wireshark/template [hunk 2]
```diff
@@ -55,18 +50,14 @@ libwireshark-devel_package() {
 	pkg_install() {
 		vmove usr/include
 		vmove usr/lib/pkgconfig
-		vmove usr/lib/*.so
+		vmove "usr/lib/*.so"
 	}
 }
 
 wireshark-gtk_package() {
```
Comment (maxice8) [inline srcpkgs/wireshark/template]: archs=noarch as well

**Comment by CameronNemo** [comment]
Cross is broken.

```
Scanning dependencies of target lemon
[  7%] Building C object tools/lemon/CMakeFiles/lemon.dir/lemon.c.o
[  7%] Linking C executable ../../run/lemon
[  7%] Built target lemon
[  7%] Generating grammar.c, grammar.h, grammar.out
/bin/sh: lemon: command not found
```

```
masterdir/builddir/wireshark-3.0.0/build/run/lemon: ELF 64-bit LSB pie executable, ARM aarch64, [...]
```

**Comment by maxice8** [comment]
On Sat, Mar 09, 2019 at 02:58:09PM -0800, Cameron Nemo wrote:
> Cross is broken.
> 
> ```
> Scanning dependencies of target lemon
> [  7%] Building C object tools/lemon/CMakeFiles/lemon.dir/lemon.c.o
> [  7%] Linking C executable ../../run/lemon
> [  7%] Built target lemon
> [  7%] Generating grammar.c, grammar.h, grammar.out
> /bin/sh: lemon: command not found
> ```
> 
> ```
> masterdir/builddir/wireshark-3.0.0/build/run/lemon: ELF 64-bit LSB pie executable, ARM aarch64, [...]
> ```
> 
> -- 
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub:
> https://github.com/void-linux/void-packages/pull/9589#issuecomment-471229843

Needs the cmake file to be modified to build lemon with host tools

**Comment by jnbr** [comment]
Here are patches to fix cross builds:

```
--- tools/lemon/CMakeLists.txt	2019-02-28 18:56:38.000000000 +0100
+++ tools/lemon/CMakeLists.txt	2019-03-11 20:41:14.211616335 +0100
@@ -7,6 +7,14 @@
 # SPDX-License-Identifier: GPL-2.0-or-later
 #

+if(CMAKE_CROSSCOMPILING)
+	add_custom_target(
+		lemon
+		COMMAND $ENV{CC_FOR_BUILD} -o ${EXECUTABLE_OUTPUT_PATH}/lemon ${CMAKE_CURRENT_SOURCE_DIR}/lemon.c
+		COMMENT "generate lemon for host"
+	)
+else()
+
 add_executable(lemon lemon.c)

 # To keep lemon.c as close to upstream as possible, deliberately ignore
@@ -62,6 +70,7 @@
 endif()
 target_compile_options(lemon PRIVATE ${lemon_cflags})

+endif() # CMAKE_CROSSCOMPILING
 #
 # Editor modelines  -  https://www.wireshark.org/tools/modelines.html
 #

--- cmake/modules/UseLemon.cmake	2019-02-28 18:56:38.000000000 +0100
+++ cmake/modules/UseLemon.cmake	2019-03-11 21:19:28.352163076 +0100
@@ -13,7 +13,7 @@
           # These files are generated as side-effect
           ${_out}.h
           ${_out}.out
-         COMMAND lemon
+         COMMAND ${EXECUTABLE_OUTPUT_PATH}/lemon
            -T${_lemonpardir}/lempar.c
            -d.
            ${_in}
```

Downside is, that `UseLemon.cmake` and `lemon` must not be packaged with these patches applied.

File: srcpkgs/wireshark/template [hunk 3]
```diff
@@ -44,7 +38,7 @@ post_install() {
 libwireshark_package() {
 	short_desc+=" - library"
 	pkg_install() {
-		vmove usr/lib/*.so.*
+		vmove "usr/lib/*.so.*"
 		vmove usr/lib/wireshark
```
Comment (jnbr) [inline srcpkgs/wireshark/template]: this installs a bunch of cmake files

**Comment by jnbr** [comment]
still WIP?

**Comment by CameronNemo** [comment]
I was going to test in a few hours before bumping this PR. If it works for you then it should be no different from the pre-crossfix.

**Comment by CameronNemo** [comment]
Working pretty well here!

**Pull request merged**